src/coreclr/vm/util.cpp

```diff
@@ -1774,24 +1774,14 @@ static BOOL TrustMeIAmSafe(void *pLock)
 
 LockOwner g_lockTrustMeIAmThreadSafe = { NULL, TrustMeIAmSafe };
 
-static DangerousNonHostedSpinLock g_randomLock;
```

Comment:
Nearly all places that use this random number generator seems to be debug-only code, except one in ceemain.cpp; `ExecutableAllocator::InitLazyPreferredRange`. It seems wasteful to carry 200+ bytes of thread statics for each thread just to make that one place work in release builds.